plugins/inputs/win_wmi/README.md

```diff
@@ -0,0 +1,229 @@
+# Windows Management Interface Input Plugin
+
+This document presents the input plugin to read WMI classes on Windows
+operating systems. With the win_wmi plugin, an administrator is enabled to
+capture and filter virtually any configuration or metric value exposed through
+the Windows Management Instrumentation service.
+
+If a WMI class property's value is a string, then the string is included with
+the metric as a tag. If a WMI class property's value is an integer, then the
+integer is used as a metric field.
+
+If telegraf is configured with a logfile and the plugin's configuration
+contains an invalid namespace, class, or property, an error is logged.
+
+## Basics
+
+The examples contained in this file have been utilized and exercised in Windows
+environments. There are many other useful classes to monitor if you know what
+to look for.
+
+### Namespace
+
+A string representing the WMI namespace to be queried. For example,
+`root\\cimv2`.
+
+### ClassName
+
+A string representing the WMI class to be queried. For example,
+`Win32_Processor`.
+
+### Properties
+
+An array of strings representing the properties of the WMI class to be queried.
+By default, the `Name` property is included in the query. However, some classes
+do not contain a `Name` property, in which cases the `ExcludeNameKey`
+configuration should be utilized.
+
+### ExcludeNameKey
+
+By default, a WMI class's `Name` property is included as a tag value in order
+for each metric to have a unique identifier. However, some WMI classes do not
+have a `Name` property. In such cases, ExcludeNameKey should be set to `True`.

```

Comment:
This can be achieved with [tagexclude](https://github.com/influxdata/telegraf/tree/master/config#metric-filtering) so to me this setting seems redundant.

You say it needs to be set when the class does not have this Name property. Does it error if this is forgotten?

Reply:
Yes, there will be an error if:
1. WMI class does not contain a `Name` property
2. `ExcludeNameKey` is `false`

On the vast majority of WMI classes, `Name` is the unique friendly identifier of the element in the class. There are a few classes which don't follow that convention. For example, `MBAM_Volume` uses `VolumeName` instead of `Name`. :( Running the query `SELECT Name FROM MBAM_Volume` will result in an `Invalid query` error.

There are perhaps two ways to eliminate the need for `ExcludeNameKey`:
1. Require the operator to specify the Name property as part of the `Properties` configuration.
2. Make an additional WMI class for each class, inspecting the class definition to determine if a `Name` property exists. 

If a change were to be made here, option 1 would be most appropriate.

Reply:
Thanks. Would also go for option one.